test/e2e/gitops_service_test.go

```diff
@@ -301,3 +306,72 @@ func validateMachineConfigUpdates(t *testing.T) {
 	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: existingImage.Name}, existingImage)
 	assertNoError(t, err)
 }
+
+func validateNonDefaultArgocdNamespaceManagement(t *testing.T) {
+	framework.AddToFrameworkScheme(argoapi.AddToScheme, &argoapp.ArgoCD{})
+	framework.AddToFrameworkScheme(configv1.AddToScheme, &configv1.ClusterVersion{})
+
+	ctx := framework.NewContext(t)
+	defer ctx.Cleanup()
+	f := framework.Global
+
+	// Create non-default argocd source namespace
+	argocdNonDefaultNamespaceObj := &corev1.Namespace{ObjectMeta: v1.ObjectMeta{Name: argocdNonDefaultNamespace}}
+	err := f.Client.Create(context.TODO(), argocdNonDefaultNamespaceObj, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	assertNoError(t, err)
+
+	// Create non-default namespace argocd instance
+	argocdNonDefaultNamespaceInstance, err := argocd.NewCR(argocdNonDefaultNamespaceInstanceName, argocdNonDefaultNamespace)
+	err = f.Client.Create(context.TODO(), argocdNonDefaultNamespaceInstance, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	assertNoError(t, err)
+
+	// Check if non-default argocd namespace is created
+	existingArgocdNonDefaultNamespaceObj := &corev1.Namespace{ObjectMeta: v1.ObjectMeta{Name: argocdNonDefaultNamespace}}
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: argocdNonDefaultNamespace}, existingArgocdNonDefaultNamespaceObj)
+	assertNoError(t, err)
+
+	// Check if non-default namepsace argocd instance is created
+	existingArgoInstance, err := argocd.NewCR(argocdNonDefaultNamespaceInstanceName, argocdNonDefaultNamespace)
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: argoCDInstanceName, Namespace: argoCDNamespace}, existingArgoInstance)
+	assertNoError(t, err)
+
+	identityProviderYAML := filepath.Join("test", "yamls", "identity-provider_appcr.yaml")
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command(ocPath, "apply", "-f", identityProviderYAML)
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = wait.Poll(time.Second*1, time.Second*60, func() (bool, error) {
+		if err := helper.ApplicationHealthStatus("identity-provider", argocdNonDefaultNamespace); err != nil {
+			t.Log(err)
+			return false, nil
+		}
+		if err := helper.ApplicationSyncStatus("identity-provider", argocdNonDefaultNamespace); err != nil {
+			t.Log(err)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Tear down non-default namespace Argo CD instance
+	// err = f.Client.Delete(context.TODO(), existingArgoInstance, &client.DeleteOptions{})
+	// assertNoError(t, err)
+
+	// err = e2eutil.WaitForDeletion(t, f.Client.Client, existingArgoInstance, retryInterval, timeout)
+	// assertNoError(t, err)
+
+	// // Delete non-default Argo CD namespace
```

Comment:
@jaideepr97 This could be helpful - https://github.com/redhat-developer/gitops-operator/pull/141/files#diff-451b2db506ec846e24c6d56da6531ff480580cd9812c2daf225cc5a1cf7fa35cR328

Reply:
Thanks Amit, had a look but didn't end up needing to use it eventually